### PR TITLE
Add first pass of ARC20 documentation

### PR DESCRIPTION
**Pull request by aaronbuchwald**

### CODE REVIEW

File: build/references/coreth-arc20s.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
+# Coreth ANTs and ARC20s
+
+## ANTs on the C Chain
+
+### Why move an ANT from the X Chain to the C Chain?
```
Comment (cgcardona) [inline build/references/coreth-arc20s.md:5]: As a convention should we use dashes when talking about chains?

> the X Chain to the C Chain.
or

> the X-Chain to the C-Chain.

File: build/references/coreth-arc20s.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+When we want to use smart contract functionality, we need a total ordering of state transitions (transactions), so that each transaction can modify the state of the network without having to worry about external dependencies.
+
+Therefore, when we want to use smart contract functionality, we must perform an atomic swap to move ANTs from the X Chain to the C Chain.
+
+### ANTs on the C Chain
+
+#### Native Coin - AVAX
+
+ANTs on the C Chain are either AVAX - the Avalanche native asset - or another asset that was created on the X Chain.
```
Comment (cgcardona) [inline build/references/coreth-arc20s.md:17]: Use proper `em` dashes w/ `&emdash;`

`ANTs on the C Chain are either AVAX&mdash;the Avalanche native asset&mdash;or another asset that was created on the X Chain.`

ANTs on the C Chain are either AVAX&mdash;the Avalanche native asset&mdash;or another asset that was created on the X Chain.

File: build/references/coreth-arc20s.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+Non-AVAX Assets, however, have no counterpart within the EVM. Therefore, Coreth makes minor modifications to the EVM in order to support asset balances and the ability to transfer these assets on the C Chain.
+
+Non-AVAX Assets (ANTs) are supported on the C Chain by keeping an account balance mapping [assetID -> balance]. These assets can be exported back to the X Chain or moved around on the C Chain using `assetCall` and `assetBalance`.
+
+##### assetCall
+
+`assetCall` is a precompiled contract at address `0xassetCallAddr`. `assetCall` allows users to atomically transfer a native asset to a given address and also make a contract call to that same address.
+
+This is identical to how a normal transaction can send value to some address, while also atomically calling that address with some `data`.
+
```
Comment (cgcardona) [inline build/references/coreth-arc20s.md:44]: Remove extra line

File: build/references/coreth-arc20s.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
++-------------+---------------+--------------------------------+
+| address     : address       |                       20 bytes |
++-------------+---------------+--------------------------------+
+| assetID     : uint256       |                       32 bytes |
++-------------+---------------+--------------------------------+
+| assetAmount : uint256       |                       32 bytes |
++-------------+---------------+--------------------------------+
+| callData    : bytes memory  |            len(callData) bytes |
++-------------+---------------+--------------------------------+
+                              |       84 * len(callData) bytes |
```
Comment (cgcardona) [inline build/references/coreth-arc20s.md:63]: This should be `84 + len(callData) bytes` and not `84 * len(callData) bytes`

Replace `*` with `+`

File: build/references/coreth-arc20s.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+                              +--------------------------------+
+```
+
+
```
Comment (cgcardona) [inline build/references/coreth-arc20s.md:67]: Remove extra line

File: build/references/coreth-arc20s.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+##### assetBalance
+
+`assetBalance` is a precompiled contract at address `0xassetBalance`. `assetBalance` is the ANT equivalent of using `balance` to get the AVAX balance.
+
```
Comment (cgcardona) [inline build/references/coreth-arc20s.md:71]: Remove extra line

File: build/references/coreth-arc20s.md [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+```text
++-------------+---------------+-----------------+
+| address     : address       |        20 bytes |
++-------------+---------------+-----------------+
+| assetID     : uint256       |        32 bytes |
++-------------+---------------+-----------------+
+                              |        52 bytes |
+                              +-----------------+
+```
+
```
Comment (cgcardona) [inline build/references/coreth-arc20s.md:88]: Remove extra line

**Review by cgcardona — CHANGES_REQUESTED** [review]
Amazing PR. Please see inline comments for requested changes. 🏅

File: build/references/coreth-arc20s.md (hunk 1 above)
Comment (aaronbuchwald) [reply build/references/coreth-arc20s.md]: Updated

File: build/references/coreth-arc20s.md (hunk 2 above)
Comment (aaronbuchwald) [reply build/references/coreth-arc20s.md]: Updated

File: build/references/coreth-arc20s.md (hunk 4 above)
Comment (aaronbuchwald) [reply build/references/coreth-arc20s.md:63]: Good catch

File: build/references/coreth-arc20s.md (hunk 3 above)
Comment (aaronbuchwald) [reply build/references/coreth-arc20s.md:44]: Updated

File: build/references/coreth-arc20s.md [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+Therefore, when we want to use smart contract functionality, we must perform an atomic swap to move ANTs from the X-Chain to the C-Chain.
+
+### ANTs on the C-Chain
+
+#### Native Coin - AVAX
+
+ANTs on the C-Chain are either AVAX &emdash; the Avalanche native asset &emdash; or another asset that was created on the X-Chain.
+
+The C-Chain is a modified instance of the EVM, so AVAX on the C-Chain is equivalent to ETH on the Ethereum network. When you create or call a smart contract, you pay for gas costs with AVAX. You can natively transfer AVAX between accounts, send it to a smart contract, all using native EVM tools and libraries.
```
Comment (dasconnor) [inline build/references/coreth-arc20s.md:19]: I think you should do Ethereum Virtual Machine (EVM) for your first use here.

File: build/references/coreth-arc20s.md [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+
+#### EVM Transaction
+
+An EVM Transaction is composed of the following fields:
+
+* **`nonce`** Scalar value equal to the number of transactions send by the sender
```
Comment (dasconnor) [inline build/references/coreth-arc20s.md:25]: sent

File: build/references/coreth-arc20s.md [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+* **`gasPrice`** Scalar value equal to tthe number of Wei paid per unit of gas to execute this transaction
```
Comment (dasconnor) [inline build/references/coreth-arc20s.md:26]: tthe

File: build/references/coreth-arc20s.md [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+* **`gasLimit`** Scalar value equal to the maximum amount of gas that should be used in executing this transaction
+* **`to`** The 20 byte address of the message call's recipient. If the transaction is creating a contract, `to` is left empty
+* **`value`** Scalar value of native asset (AVAX) measured in wei to be transferred to the message call's recipient or in the case of a contract creation, as an endowment to the newly created contract.
+* **`v, r, s`** Values corresponding to the signature of the transaction.
```
Comment (dasconnor) [inline build/references/coreth-arc20s.md:30]: I prefer initialization for the American spelling, but up to you.

File: build/references/coreth-arc20s.md [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
++-------------+---------------+-----------------+
+                              |        52 bytes |
+                              +-----------------+
+```
+
+## ARC20s on the C-Chain
+
+An ARC20 is the Avalanche equivalent of an ERC20. An ERC20 is just a smart contract that maintains balances for accounts and provides an incredibly useful, standardized interface for smart contract developers.
+
+ERC20s can be deployed directly to the C-Chain without modification, but these contracts do not have a native asset equivalent. Therefore, we create `Wrapped ARC20s` (`WARC20s`) which wrap native assets into an ERC20 interface, so that `ANTs` can be used following the same interface as an ERC20.
```
Comment (dasconnor) [inline build/references/coreth-arc20s.md:89]: I think you got everything right here, but I would love a Confluence page that outlines our style guide for ERC20 vs ARC20 usage. Also, technically it is ERC-20.

**Review by dasconnor — CHANGES_REQUESTED** [review]

**Comment by dasconnor** [comment]
Nice work!

File: build/references/coreth-arc20s.md (hunk 10 above)
Comment (danlaine) [reply build/references/coreth-arc20s.md:26]: Do we define how many wei == 1 nAVAX?

File: build/references/coreth-arc20s.md [hunk 13]
```diff
@@ -0,0 +1,103 @@
...
+* **`gasPrice`** Scalar value equal to tthe number of Wei paid per unit of gas to execute this transaction
+* **`gasLimit`** Scalar value equal to the maximum amount of gas that should be used in executing this transaction
+* **`to`** The 20 byte address of the message call's recipient. If the transaction is creating a contract, `to` is left empty
+* **`value`** Scalar value of native asset (AVAX) measured in wei to be transferred to the message call's recipient or in the case of a contract creation, as an endowment to the newly created contract.
+* **`v, r, s`** Values corresponding to the signature of the transaction.
+* **`data`** Unlimited size byte array specifying the input data to a contract call or (if creating a contract) the EVM bytecode for the account initialisation process
+
+#### Native Assets - Tokens
+
+Non-AVAX Assets, however, have no counterpart within the EVM. Therefore, Coreth makes minor modifications to the EVM and state management in order to support asset balances and the ability to transfer these assets on the C-Chain.
```
Comment (danlaine) [inline build/references/coreth-arc20s.md:35]: I'd rephrase to `Non-AVAX Assets, however, have no native representation on the EVM.`

File: build/references/coreth-arc20s.md (hunk 10 above)
Comment (aaronbuchwald) [reply build/references/coreth-arc20s.md:26]: 1 wei = 1 nAVAX * 10^-9. I think it will be better to use the wei convention here because atm that is how all of the tools will display the information. I'll add in a conversion though.

File: build/references/coreth-arc20s.md (hunk 11 above)
Comment (aaronbuchwald) [reply build/references/coreth-arc20s.md:30]: I do too, just borrowed these from a writeup on Ethereum transactions, I'll update it.

File: build/references/coreth-arc20s.md [hunk 14]
```diff
@@ -0,0 +1,103 @@
...
++-------------+---------------+-----------------+
+| assetID     : uint256       |        32 bytes |
++-------------+---------------+-----------------+
+                              |        52 bytes |
+                              +-----------------+
+```
+
+## ARC20s on the C-Chain
+
+An ARC20 is the Avalanche equivalent of an ERC20. An ERC20 is just a smart contract that maintains balances for accounts and provides an incredibly useful, standardized interface for smart contract developers.
```
Comment (mussallem) [inline build/references/coreth-arc20s.md:87]: Recommend removing "just" to make tone neutral.

"An ERC20 is **just** a smart contract that maintains balances" to "n ERC20 is a smart contract that maintains balances"

File: build/references/coreth-arc20s.md (hunk 12 above)
Comment (aaronbuchwald) [reply build/references/coreth-arc20s.md:89]: Ok, I'll change ERC20 -> ERC-20 and ARC20 -> ARC-20

File: build/references/coreth-arc20s.md (hunk 14 above)
Comment (aaronbuchwald) [reply build/references/coreth-arc20s.md]: Updated

Comment (aaronbuchwald) [reply build/references/coreth-arc20s.md]: It's difficult to describe the difference without using some modifier. I changed it to describing ERC-20s as second class citizens in the world of Ethereum (because they are stored on the smart contract instead of on the account that owns them), but added the qualifier that the rich interface they create enables the powerful world of DeFi.

File: build/references/coreth-arc20s.md [hunk 15]
```diff
@@ -0,0 +1,136 @@
...
+
+When we want to use smart contract functionality, we need a total ordering of state transitions (transactions), so that each transaction can modify the state of the network without having to worry about external dependencies.
+
+Therefore, when we want to use smart contract functionality, we must perform an atomic swap to move ANTs from the X-Chain to the C-Chain.
+
+### ANTs on the C-Chain
+
+#### Native Coin - AVAX
+
+ANTs on the C-Chain are either AVAX &emdash; the Avalanche native asset &emdash; or another asset that was created on the X-Chain.
```
Comment (ApolloUnicorn) [inline build/references/coreth-arc20s.md:17]: Looks like the emdash isn't showing up the way it's intended

File: build/references/coreth-arc20s.md [hunk 16]
```diff
@@ -0,0 +1,136 @@
...
+* **`to`** The 20 byte address of the message call's recipient. If the transaction is creating a contract, `to` is left empty
+* **`value`** Scalar value of native asset (AVAX) measured in Wei (1 Wei = 10^-18 AVAX) to be transferred to the message call's recipient or in the case of a contract creation, as an endowment to the newly created contract.
+* **`v, r, s`** Values corresponding to the signature of the transaction.
+* **`data`** Unlimited size byte array specifying the input data to a contract call or (if creating a contract) the EVM bytecode for the account initialization process
+
+#### Native Assets - Tokens
+
+Non-AVAX Assets, however, have no native representation within the EVM. Therefore, Coreth makes minor modifications to the EVM and state management in order to support asset balances and the ability to transfer these assets on the C-Chain.
+
+Non-AVAX Assets (ANTs) are supported on the C-Chain by keeping an account balance mapping [assetID -> balance]. These assets can be exported back to the X-Chain or moved around on the C-Chain using `assetCall` and `assetBalance`.
```
Comment (ApolloUnicorn) [inline build/references/coreth-arc20s.md:35]: Are non-AVAX assets also called ANT's?

Comment (aaronbuchwald) [reply build/references/coreth-arc20s.md:35]: Yes, ANTs are Avalanche Native Tokens, so if you mint an asset on the X-Chain then it is an ANT.

File: build/references/coreth-arc20s.md (hunk 15 above)
Comment (aaronbuchwald) [reply build/references/coreth-arc20s.md]: Thanks! Just updated

File: build/references/coreth-arc20s.md [hunk 17]
```diff
@@ -0,0 +1,165 @@
...
+ERC-20s typically have a total supply field, but this can mean different things in the context of a wrapped asset. The total supply could indicate the total supply of the non-wrapped asset on the entire platform or the amount of the asset in the wrapper contract.
+
+For simplicity, we will use total supply to indicate the total supply of the wrapped asset in the ARC-20 contract.
+
+#### ARC-20 Deposits
+
+Similar to WETH, in order to deposit some funds into an ARC-20, we need to send the ARC-20 contract the deposit amount and then invoke the contract's deposit function, so that the contract can acknowledge the deposit and update the caller's balance. With WETH, this can be accomplished with a simple `call` because it allows the caller to both send the native coin and invoke a smart contract. With non-AVAX ARC-20s, `assetCall` allows the same functionality for all ANTs on the C-Chain.
+
+* **`nonce`**: 2
+* **`gasPrice`**: 470 gwei
```
Comment (aphexmunky) [inline build/references/coreth-arc20s.md:126]: outdated gasPrice value
